pkg/chunk/cassandra/storage_client.go

```diff
@@ -116,10 +117,23 @@ func (cfg *Config) session() (*gocql.Session, error) {
 func (cfg *Config) setClusterConfig(cluster *gocql.ClusterConfig) error {
 	cluster.DisableInitialHostLookup = cfg.DisableInitialHostLookup
 
+	serverNames := strings.Split(cfg.Addresses, ",")
 	if cfg.SSL {
-		cluster.SslOpts = &gocql.SslOptions{
-			CaPath:                 cfg.CAPath,
-			EnableHostVerification: cfg.HostVerification,
+		if cfg.HostVerification {
+			if len(serverNames) != 1 {
+				return errors.New("host verification is only possible for a single host")
+			}
```

Comment:
Could you move this check to `cfg.Validate()`? It's based on `cfg.Addresses` and `cfg.HostVerification` so should be easy to do.

Reply:
Done!